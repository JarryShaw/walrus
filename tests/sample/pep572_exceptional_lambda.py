import re

(lambda: ((x := 99), print(x)))()  # Valid, but unlikely to be useful
print(globals().get('x'))
(lambda: ([x := i ** 2 for i in range(10)], print(x)))()  # comprehension in lambda
print(globals().get('x'))
(x := lambda: 111)  # Valid
print(x())
print((lambda line: (m := re.match(r'(\d+) (\d+)', line)) and m.group(1))('347 467'))  # Valid
print(globals().get('m'))
