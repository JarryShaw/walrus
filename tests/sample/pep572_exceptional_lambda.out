99
None
81
None
111
347
None
